odl/trafos/fourier.py

```diff
@@ -0,0 +1,1024 @@
+            raise ValueError('Expected input shape {}, got {}.'
+                             ''.format(tuple(in_shape), arr_in.shape))
+
+        if is_real_dtype(arr_out.dtype):
+            in_dtype = _TYPE_MAP_R2C[arr_out.dtype]
+        elif halfcomplex:
+            raise ValueError('Cannot combine halfcomplex backward transform '
+                             'with complex output.')
+        else:
+            in_dtype = arr_out.dtype
+
+        if arr_in.dtype != in_dtype:
+            raise TypeError('Expected input dtype {}, got {}.'
+                            ''.format(in_dtype, arr_in.dtype))
+
+    else:  # Shouldn't happen
+        raise RuntimeError
+
+
+def pyfftw_call(array_in, array_out, direction='forward', axes=None,
+                halfcomplex=False, **kwargs):
+    """Calculate the DFT.
+
+    The discrete Fourier transform calcuates the sum
+
+        :math:`\widehat{f}_k =
+        \sum_{j \\in I_N} f_j\ e^{-i 2\pi j\odot k / N}`
+
+    for indices :math:`k \\in I_N` or, in the half-complex case,
+    :math:`0 \\leq k_d \\leq \\lfloor N_d / 2 \\rfloor + 1` for the
+    last component.
+
+    Parameters
+    ----------
+    array_in : `numpy.ndarray`
+        Array to be transformed
+    array_out : `numpy.ndarray`
+        Output array storing the transformed values
+    direction : {'forward', 'backward'}
+        Direction of the transform
+    axes : sequence of `int`, optional
+        Dimensions along which to take the transform. `None` means
+        using all axis and is equivalent to ``np.arange(ndim)``.
+    halfcomplex : `bool`, optional
+        If `True`, calculate only the negative frequency part along the
+        last axis. If `False`, calculate the full complex FFT.
+        This option can only be used with real input data.
+    fftw_plan : ``pyfftw.FFTW``, optional
+        Use this plan instead of calculating a new one. If specified,
+        the options ``planning_effort``, ``planning_timelimit`` and
+        ``threads`` have no effect.
+    planning_effort : {'estimate', 'measure', 'patient', 'exhaustive'}
+        Flag for the amount of effort put into finding an optimal
+        FFTW plan. See the `FFTW doc on planner flags
+        <http://www.fftw.org/fftw3_doc/Planner-Flags.html>`_.
+    planning_timelimit : `float`, optional
+        Limit planning time to roughly this amount of seconds.
+        Default: `None` (no limit)
+    threads : `int`, optional
+        Number of threads to use. Default: 1
+    normalise_idft : `bool`, optional
+        If `True`, the backward transform is normalized by
+        ``1 / N``, where ``N`` is the total number of points in
+        ``array_in[axes]``. This ensures that the IDFT is the true
+        inverse of the forward DFT.
+        Default: `False`
+    import_wisdom : `str`, optional
+        File name to load FFTW wisdom from. If the file does not exist,
+        it is ignored.
+    export_wisdom : `str`, optional
```

Comment:
I think we should follow the `numpy.savetxt` pattern here:

> fname : filename or file handle

That way users could provide a file handle for some file in an arbitrary folder.